code/game/objects/items/stacks/medical.dm

```diff
@@ -147,6 +147,27 @@
 	stack_id = "Tear petals"
 
 
+/obj/item/stack/medical/advanced/
+	dir = NORTH
+	flags_atom = DIRLOCK
+
+/obj/item/stack/medical/advanced/update_icon()
+	if(max_amount < 1)
+		return
+	var/percentage = round(amount / max_amount) * 100
+	if(amount >= max_amount)
```

Comment:
Early return, combine it with the first check.

Reply:
Is the check even necessary? As it it, it should never be true, so this should only work for full stacks.
If you wanted it not to update the sprite if the amount is higher than the maximum possible, the why set the switch to account for `to INFINITY`?